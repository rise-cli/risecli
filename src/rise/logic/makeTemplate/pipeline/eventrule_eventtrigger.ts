export function makeEventRule({
    apiName,
    eventBus,
    eventSource,
    eventName,
    index
}: {
    apiName: string
    eventName: string
    index: number
    eventBus: string
    eventSource: string
}) {
    return {
        Resources: {
            [`EventListener${apiName}${eventName}${index}`]: {
                Type: 'AWS::Events::Rule',
                Properties: {
                    EventBusName: eventBus,
                    EventPattern: {
                        source: [`custom.${eventSource}`],
                        'detail-type': [eventName]
                    },
                    Targets: [
                        {
                            Arn: {
                                'Fn::GetAtt': [
                                    `TriggerSubscriptionFunction${apiName}${eventName}${index}`,
                                    'Arn'
                                ]
                            },
                            Id: `EventListener${apiName}${eventName}${index}`
                        }
                    ]
                }
            },

            [`EventRuleRole${apiName}${eventName}${index}`]: {
                Type: 'AWS::Lambda::Permission',
                Properties: {
                    FunctionName: {
                        'Fn::GetAtt': [
                            `TriggerSubscriptionFunction${apiName}${eventName}${index}`,
                            'Arn'
                        ]
                    },
                    Action: 'lambda:InvokeFunction',
                    Principal: 'events.amazonaws.com',
                    SourceArn: {
                        'Fn::GetAtt': [
                            `EventListener${apiName}${eventName}${index}`,
                            'Arn'
                        ]
                    }
                }
            }
        },
        Outputs: {}
    }
}
