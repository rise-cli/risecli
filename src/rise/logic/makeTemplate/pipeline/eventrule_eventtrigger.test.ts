import { makeEventRule } from './eventrule_eventtrigger'

test('makeEventRule will render valid cloudformation', () => {
    const result = makeEventRule({
        apiName: 'apiName',
        eventName: 'eventName',
        index: 2,
        eventBus: 'eventBus',
        eventSource: 'eventSource'
    })

    expect(result).toMatchSnapshot()
})
