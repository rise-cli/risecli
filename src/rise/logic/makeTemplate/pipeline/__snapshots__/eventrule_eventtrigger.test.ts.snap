// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeEventRule will render valid cloudformation 1`] = `
Object {
  "Outputs": Object {},
  "Resources": Object {
    "EventListenerapiNameeventName2": Object {
      "Properties": Object {
        "EventBusName": "eventBus",
        "EventPattern": Object {
          "detail-type": Array [
            "eventName",
          ],
          "source": Array [
            "custom.eventSource",
          ],
        },
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "TriggerSubscriptionFunctionapiNameeventName2",
                "Arn",
              ],
            },
            "Id": "EventListenerapiNameeventName2",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "EventRuleRoleapiNameeventName2": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "TriggerSubscriptionFunctionapiNameeventName2",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "EventListenerapiNameeventName2",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
}
`;
